### PR TITLE
feat(edgeless): update frame ui

### PR DESCRIPTION
**Pull request by L-Sun**
### TL;DR

Improved frame UI in the edgeless mode. [BS-1112](https://linear.app/affine-design/issue/BS-1112/[frame-and-group-improvement]-ui-update)

### What changed?
- UI changes:
  - Colorize frame title background and reverse text color based on it. [BS-1114](https://linear.app/affine-design/issue/BS-1114/[frame-improvement]-frame-标签文字颜色需要根据-color-picker-选定的颜色自动反色)
  - Hide frame title when frame is small or text is empty. [BS-1111](https://linear.app/affine-design/issue/BS-1111/[frame-improvement]-标签的宽度受-frame-宽度的影响，标签内容进行省略处理，缩放直到内容全部省略后消失), [BS-1306](https://linear.app/affine-design/issue/BS-1306/[ui]-使用空格-hack-标签，标签不渲染)
  - Highlight frame border when hover frame title. [BS-1110](https://linear.app/affine-design/issue/BS-1110/[frame-improvement]-frame-通过-hover-标签进入-hover-态)
  - Mask frame title when hover it. [BS-1307](https://linear.app/affine-design/issue/BS-1307/[ui]-hover-标签时应该有-000000-透明度的黑色的-hover-态)
  - Highlight child elements when resize frame.
  - Update style of dragging rect. [BS-1294](https://linear.app/affine-design/issue/BS-1294/[improvement]-修改-edgeless-鼠标选区的-ui)
- UI bug fixes:
  - Fixed incorrect border of frame in `<affine-surface-ref>` in page mode. [BS-1217](https://linear.app/affine-design/issue/BS-1217/embed-frame-出现双边框)
  - Fixed jitter when enter frame title editor. [BS-1304](https://linear.app/affine-design/issue/BS-1304/[bug]-frame-标签-rename-时抖动)
  - Fixed viewport of <affine-surface-ref> not changed after resize frame. [BS-1212](https://linear.app/affine-design/issue/BS-1212/通过peekview调整frame的大小后，文档中的frame的viewport没有更新)

### CODE REVIEW

**Comment by L-Sun** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/toeverything/blocksuite/8108?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#8108** <a href="https://app.graphite.dev/github/pr/toeverything/blocksuite/8108?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#8048** <a href="https://app.graphite.dev/github/pr/toeverything/blocksuite/8048?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @L-Sun and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by doodlewind — APPROVED** [review]
